src/frontend/src/handler/create_table.rs

```diff
@@ -46,71 +46,76 @@ use crate::stream_fragmenter::build_graph;
 /// If a column is marked as `primary key`, its `ColumnId` is also returned.
 /// This primary key is not combined with table constraints yet.
 pub fn bind_sql_columns(columns: Vec<ColumnDef>) -> Result<(Vec<ColumnDesc>, Option<ColumnId>)> {
+    bind_sql_columns_with_offset(columns, 0)
+}
+
+/// Bind the columns with an `offset` on the column IDs to allocate. See [`bind_sql_columns`].
+pub fn bind_sql_columns_with_offset(
+    columns: Vec<ColumnDef>,
+    column_id_offset: i32,
+) -> Result<(Vec<ColumnDesc>, Option<ColumnId>)> {
```

Comment:
Actually, the column ID here is not related to those generated by `StreamMaterialize` and stored in the table catalog. Further refactor of managed column IDs is required.